docs/DeveloperGuide.md

```diff
@@ -196,10 +196,131 @@ H  I     J  K      L  M      N  O
 ```
 
 
+
+
+## Implementation of Logic Gate Commands
+
+There are six different logic gates that can be instantiated in the program:
+- `or, and, nor, nand, xor, xnor`
+
+All gates can be first instantiated using the `Gate` class which has one `int input` and one `int output` as its attributes. 
+It has `setInput(int input)` and `getOutput()` as its methods, which are used to set the input of the logic gate and get the output of the gate respectively. 
+
+The six different logic gates take in two different inputs, which necessitates the need for a logic gate class to take in one more input.
+This can be achieved by instantiating the `TwoInputGate` class which inherits from the `Gate` class. It has an additional attribute `int secondInput`
+to take in the additional input and a method `setSecondInput(int input)` to set the Boolean value of either `0 or 1` to the additional input.
+
+The six different logic gates then individually inherit from the `TwoInputGate` class where the return value of their `getOutput()` function depends
+on the logic function of the gate itself. For instance, the `OrGate` which inherits from the `TwoInputGate` class has its `getOutput()` function set to `{return input | secondInput}`, 
+which represents the `or` operation.
+
+The `OrGate` can be visualised as such in the following object diagram:
+
+![InsertObjectDiagram](diagrams/OrGateObjectDiagram.png)
+ 
+The inheritance of the `OrGate` class from `TwoInputGate` class which inherits from the `Gate` class can be seen in the following class diagram:
+
+![InsertClassDiagram](diagrams/OrGateClassDiagram.png)
+
+
+There are four Boolean commands that are used in the implementation of the logic gates: `TemplateBooleanCommand, SetBooleanCommand, AddBooleanCommand`, and `CalcBooleanCommand`.
+#### `TemplateBooleanCommand`
+The `TemplateBooleanCommand` creates a Boolean template of any one of the six available logic gates. 
+
+The sequence by which the `TemplateBooleanCommand` is instantiated is as follows using the user input `template and` who wants to instantiate an `and` logic template.
+
+1. The function call `parse(String template)` to `BooleanParser` takes in a `String` format of the Boolean template as such: `"template and"`.
```

Comment:
Maybe these steps here and the next command can just be shown by UML diagrams mostly? If the sequence diagram is very clear, the steps taken might even be redundant.